chain/xrp/client/client.go

```diff
@@ -0,0 +1,992 @@
+}
+
+// FetchBalance fetches token balance for a XRP address
+func (client *Client) FetchBalance(ctx context.Context, address xc.Address) (xc.AmountBlockchain, error) {
+	return client.FetchBalanceForAsset(ctx, address, client.Asset)
+}
+
+func (client *Client) FetchBalanceForAsset(ctx context.Context, address xc.Address, assetCfg xc.ITask) (xc.AmountBlockchain, error) {
+	switch asset := assetCfg.(type) {
+	case *xc.ChainConfig:
+		return client.FetchNativeBalance(ctx, address)
+	case *xc.TokenAssetConfig:
+		return client.fetchContractBalance(ctx, address, asset.Contract)
+	default:
+		contract := asset.GetContract()
+		logrus.WithFields(logrus.Fields{
+			"chain":      asset.GetChain().Chain,
+			"contract":   contract,
+			"asset_type": fmt.Sprintf("%T", asset),
+		}).Warn("fetching balance for unknown asset type")
+		return client.fetchContractBalance(ctx, address, contract)
+	}
+}
+
+// FetchNativeBalance fetches account native balance for a XRP address
+func (client *Client) FetchNativeBalance(ctx context.Context, address xc.Address) (xc.AmountBlockchain, error) {
+	zero := xc.NewAmountBlockchainFromUint64(0)
+
+	request := AccountInfoRequest{
+		Method: "account_info",
+		Params: []AccountInfoParamEntry{
+			{
+				Account:     address,
+				LedgerIndex: Validated,
+			},
+		},
+	}
+
+	var accountInfoResponse AccountInfoResponse
+	err := client.Send(MethodPost, request, &accountInfoResponse)
+	if err != nil {
+		return zero, err
+	}
+
+	balance := accountInfoResponse.Result.AccountData.Balance
+	if balance == "" {
+		return zero, fmt.Errorf("empty balance returned for account: %s", address)
+	}
+
+	return xc.NewAmountBlockchainFromStr(balance), nil
+}
+
+// fetchContractBalance fetches a specific token balance based on received contract for an XRP address
+func (client *Client) fetchContractBalance(ctx context.Context, address xc.Address, assetContract string) (xc.AmountBlockchain, error) {
+	zero := xc.NewAmountBlockchainFromUint64(0)
+
+	asset, contract, err := tx.ExtractAssetAndContract(assetContract)
+	if err != nil {
+		return zero, fmt.Errorf("failed to parse and extract asset and contract: %w", err)
+	}
+
+	request := AccountLinesRequest{
+		Method: "account_lines",
+		Params: []AccountLinesParamEntry{
+			{
+				Account: address,
+			},
+		},
+	}
+
+	var accountLinesResponse AccountLinesResponse
+	err = client.Send(MethodPost, request, &accountLinesResponse)
+	if err != nil {
+		return zero, err
+	}
+
+	var balance string
+	for _, line := range accountLinesResponse.Result.Lines {
+		if line.Currency == asset && line.Account == contract {
+			balance = line.Balance
+		}
+	}
+
+	if balance == "" {
+		return zero, fmt.Errorf("empty balance returned for account: %s", address)
+	}
+
+	humanReadbleBalance, err := xc.NewAmountHumanReadableFromStr(balance)
+	if err != nil {
+		return zero, fmt.Errorf("failed to parse balance for account: %s", address)
+	}
+	return humanReadbleBalance.ToBlockchain(15), nil
```

Comment:
please define a constant `TRUSTLINE_DECIMALS` and reference that in place of `15` literals everywhere.